contracts/wrapper/PCOWrapper.sol

```diff
@@ -0,0 +1,108 @@
+// contracts/Wrapper.sol
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "../token/PartialCommonOwnership721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+
+struct WrappedToken {
+  address originAddress;
+  uint256 originId;
+  uint256 price;
+}
+
+/// @title Wrapper
+/// @author Will Holley, Victor Sint Nicolaas
+/// @notice This contract can wrap or hold other tokens adhering to the ERC721
+//          standard, and is partially common owned (see PartialCommonOwnership721).
+contract Wrapper is PartialCommonOwnership721 {
+  /// @notice Mapping tokenIds to WrappedToken's
+  mapping(uint256 => WrappedToken) private tokenMap;
+
+  // @notice Event when acquire() is called
+  event Acquire(uint256 tokenId);
+
+  constructor()
+    PartialCommonOwnership721("Partial Common Ownership Token Wrapper", "wPCO")
+  {}
+
+  /// @notice helper function to get wrapperTokenId: hash the original NFT
+  ///         contract address and its respective local tokenId
+  /// @param assetContract Address the contract defining the token
+  /// @param tokenId Token Id to be wrapped
+  function createWrappedTokenId(address assetContract, uint256 tokenId)
+    private
+    pure
+    returns (uint256)
+  {
+    return uint256(bytes32(keccak256(abi.encode(assetContract, tokenId))));
+  }
+
+  /// @notice
+  /// @param operator Address responsible for transferring the token
+  /// @param from Address the owner of the token
+  /// @param tokenId Token Id to be wrapped
+  /// @param _data extra data - unused
+  function onERC721Received(
```

Comment:
TODO: Add a comment to the docstring that indicates this is only callable via `#acquire` (and confirm this in the tests)